mathoxide-lib/src/array/binops.rs

```diff
@@ -0,0 +1,78 @@
+use num_traits::Num;
+
+use crate::array::Array;
+use crate::storage::Storage;
+use crate::thread_unsafe_storage::ThreadUnsafeStorage;
+use crate::utils::IndexIteration;
+use crate::views::{ArrayView, ContiguousView};
+
+impl<Stored, StorageType, ViewType, RhsStorage, RhsStored, RhsViewType, OutputStored>
+    std::ops::Add<&Array<RhsStorage, RhsViewType>> for &Array<StorageType, ViewType>
+where
+    RhsStorage: Storage<Stored = RhsStored>,
+    RhsStored: Num,
+    RhsViewType: ArrayView,
+    Stored: Num + std::ops::Add<RhsStored, Output = OutputStored>,
```

Comment:
Remove unnecessary traits

Reply:
the `Num` bound is needed after all, since we use methods of the `Array` type defined only when `Stored` is `Num`.